src/sensor/sensor.cpp

```diff
@@ -105,4 +108,60 @@ void Sensor::connectLinkLog(const LinkConfiguration& logConf)
     emit linkLogUpdate();
 }
 
+void Sensor::setControlPanel(const QUrl& url)
+{
+    if(!url.isValid()) {
+        qCWarning(PING_PROTOCOL_SENSOR) << "Invalid url:" << url;
+        _controlPanelUrl = _defaultControlPanelUrl;
+        return;
+    }
+
+    _controlPanelUrl = url;
+}
+
+void Sensor::setSensorVisualizer(const QUrl& url)
+{
+    if(!url.isValid()) {
+        qCWarning(PING_PROTOCOL_SENSOR) << "Invalid url:" << url;
+        //TODO: Need default visualizer
+        _sensorVisualizerUrl.setUrl({});
+        return;
+    }
+
+    _sensorVisualizerUrl = url;
+}
+
+QQuickItem* Sensor::controlPanel(QObject* parent)
+{
+    QQmlEngine* engine = qmlEngine(parent);
```

Comment:
Can we get comments or a reference here? Explaining what this does and how.

Reply:
This is a consequence of `QQmlComponent` constructor, since it's need an engine to create the component, and we are going to use the parent to use qml memory management to take care of that for us, we can extract the QQmlEngine from this parent (avoiding unnecessary methods and thread problems) to create our own component.